src/codeu/chat/client/simplegui/SignInPanel.java

```diff
@@ -44,15 +47,16 @@ private void initialize() {
         JButton signInButton = new JButton("Sign In");
         signInButton.addActionListener(new ActionListener() {
             public void actionPerformed(ActionEvent ae) {
-                HashMap<String, char[]> map = codeu.chat.UserMap.map;
+                //HashMap<String, char[]> map = codeu.chat.UserMap.map;
                 String userName = usernameField.getText();
-                if(map.containsKey(userName)) {
-                    new Conversation();
+                if(!(context.user.showUserInfo(userName).equals("Null user"))){
```

Comment:
you don't really need the extra set of parens:
if (!context.user.showUserInfo(userName).equals....)

Also, it's better to create a constant in UserContext such as
public static final String NULL_USER = "Null user"
and have context return NULL_USER in showUserInfo the username doesn't exist. Also check against NULL_USER here